lib/lte_link_control/lte_lc.c

```diff
@@ -744,8 +702,7 @@ static int connect_lte(bool blocking)
 		goto exit;
 	}
 
-	if (lte_lc_sys_mode != LTE_LC_SYSTEM_MODE_DEFAULT &&
-	    IS_ENABLED(CONFIG_LTE_NETWORK_USE_FALLBACK)) {
+	if (lte_lc_sys_mode != LTE_LC_SYSTEM_MODE_DEFAULT) {
 		/* Set system mode, because it may have been changed by fallback. */
 		if (original_func_mode != LTE_LC_FUNC_MODE_POWER_OFF &&
 		    original_func_mode != LTE_LC_FUNC_MODE_OFFLINE) {
```

Comment:
@tokangas, Need to discuss what to do with this, i.e., what is correct behavior here.

Reply:
This whole branch has been removed after offline discussion.